src/libsync/syncengine.cpp

```diff
@@ -622,7 +623,18 @@ void SyncEngine::slotDiscoveryFinished()
         return;
     }
 
-    qCInfo(lcEngine) << "#### Discovery end #################################################### " << _stopWatch.addLapTime(QLatin1String("Discovery Finished")) << "ms";
+    qCInfo(lcEngine) << "#### Discovery end #################################################### Discovery Finished" << _stopWatch.addLapTime(QLatin1String("Discovery Finished")) << "ms" << QThread::currentThread();
+    QFuture<void> future = QtConcurrent::run([=]() {
```

Comment:
The only thing this captures is `this` right? That means this could crash if SyncEngine is deleted while the thread is running.

Capture a copy of `_syncItems`, sort that, and make that the future result.

The call to `invokeMethod` is also unsafe - again, `this` could be gone. Connect the future finished signal to the continuation slot.